azurelinuxagent/ga/exthandlers.py

```diff
@@ -476,20 +479,45 @@ def handle_enable(self, ext_handler_i):
             if old_ext_handler_i is None:
                 ext_handler_i.install()
             elif ext_handler_i.version_ne(old_ext_handler_i):
-                old_ext_handler_i.disable()
-                ext_handler_i.copy_status_files(old_ext_handler_i)
-                if ext_handler_i.version_gt(old_ext_handler_i):
-                    ext_handler_i.update()
-                else:
-                    old_ext_handler_i.update(version=ext_handler_i.ext_handler.properties.version)
-                old_ext_handler_i.uninstall()
-                old_ext_handler_i.remove_ext_handler()
-                ext_handler_i.update_with_install()
+                ExtHandlersHandler._update_extension_handler(old_ext_handler_i, ext_handler_i)
```

Comment:
Thanks for this small refactoring, makes the intention of the code clearer... just a question: why a static method?

Reply:
Since I'm passing both the old_handler_i and new_handler_i there was no point making it a member function. The options were either make it a global function or make it a static member function of ExtHandlersHandler class. 

I'm open to any other design suggestion if you have an idea.

Reply:
no, this makes sense, thanks